src/interop/js-interop/usage.md

```diff
@@ -0,0 +1,368 @@
+}
+```
+
+Calling `method()` will result in a JS invocation of `library1.method()`,
+calling `JSType()` will result in a JS invocation of `new library1.JSType()`,
+and calling `JSType.staticMember` will result in a JS invocation of
+`library1.JSType.staticMember`.
+
+Unlike interop members and interop types, we only ever add a library name in the
+JS invocation if you provide a non-empty value in the `@JS()` annotation on the
+library.
+
+```dart
+library interop_library;
+
+import 'dart:js_interop';
+
+@JS()
+external void method();
+```
+
+Calling `method()` will result in a JS invocation of `method()` and *not*
+`interop_library.method()`.
+
+You can also write multiple namespaces delimited by a '.' for libraries,
+top-level members, and interop types:
+
+```dart
+@JS('library1.library2')
+library;
+
+import 'dart:js_interop';
+
+@JS('library3.method')
+external void method();
+
+@JS('library3.JSType')
+extension type JSType(JSObject _) implements JSObject {
+  external JSType();
+}
+```
+
+Calling `method()` will result in a JS invocation of
+`library1.library2.library3.method()`, calling `JSType()` will result in a JS
+invocation of `new library1.library2.library3.JSType()`, and so forth.
+
+You can't use `@JS()` annotations with '.' in the value on interop type members
+or extension members of interop types, however.
+
+If there is no value provided to `@JS()` or the value is empty, no renaming will
+occur.
+
+`@JS()` also tells the compiler that a member or type is intended to be treated
+as a JS interop member or type. It is required (with or without a value) for all
+top-level members to distinguish them from other `external` top-level members,
+but can often be elided on and within interop types and on extension members as
+the compiler can tell from the representation type and on-type.
+
+## `dart:js_interop` and `dart:js_interop_unsafe`
+
+[`dart:js_interop`] contains all the necessary members you should need,
+including `@JS`, JS types, conversion functions, and various utility functions.
+Utility functions include:
+
+- [`globalContext`], which represents the global namespace that the compilers
+  use to generate JS invocations.
+- [Helpers to type-check JS values]
+- JS operators
+- [`dartify`] and [`jsify`], which type-check and auto-convert certain JS values
+  to Dart values and vice versa.
+
+and more. More utilities might be added to this library in the future.
+
+[`dart:js_interop_unsafe`] contains members that allow you to look up properties
+dynamically. For example:
+
+```dart
+JSFunction f = console['log'];
+```
+
+Instead of declaring an interop member named `log`, we're instead using a string
+to represent the property. `dart:js_interop_unsafe` provides functionality to
+dynamically get, set, and call properties.
+
+{{site.alert.warn}}
+Avoid using `dart:js_interop_unsafe` if possible. It makes security compliance
+more difficult to guarantee and may lead to violations.
+{{site.alert.end}}
+
```

Comment:
It may be worth adding to this document also a section to read more about conversions.

In terms of usage, I think it will be good to highlight here that:
* you can use primitive or interop types at API signatures
* Callers may need to use conversions when passing values
* Some values always require an explicit conversion due to ensure correctness and consistency. This includes Lists and Functions.